packages/apollo-cache-control/src/index.ts

```diff
@@ -49,164 +47,148 @@ declare module 'apollo-server-types' {
-    // engineproxy-aimed extensions. Specifically, we want users of
-    // apollo-server-plugin-response-cache to be able to specify
-    // `cacheControl: {defaultMaxAge: 600}` without accidentally turning on the
-    // extension formatting.
-    if (this.options.stripFormattedExtensions !== false) return;
-
-    return [
-      'cacheControl',
-      {
-        version: 1,
-        hints: Array.from(this.hints).map(([path, hint]) => ({
-          path: [...responsePathAsArray(path)],
-          ...hint,
-        })),
+    return {
+      executionDidStart: () => ({
+        executionDidEnd: () => setOverallCachePolicyWhenUnset(),
+        willResolveField({ info }) {
+          let hint: CacheHint = {};
+
+          // If this field's resolver returns an object or interface, look for
+          // hints on that return type.
+          const targetType = getNamedType(info.returnType);
+          if (
+            targetType instanceof GraphQLObjectType ||
+            targetType instanceof GraphQLInterfaceType
+          ) {
+            if (targetType.astNode) {
+              hint = mergeHints(
+                hint,
+                cacheHintFromDirectives(targetType.astNode.directives),
+              );
+            }
+          }
+
+          // Look for hints on the field itself (on its parent type), taking
+          // precedence over previously calculated hints.
+          const fieldDef = info.parentType.getFields()[info.fieldName];
+          if (fieldDef.astNode) {
+            hint = mergeHints(
+              hint,
+              cacheHintFromDirectives(fieldDef.astNode.directives),
+            );
+          }
+
+          // If this resolver returns an object or is a root field and we haven't
+          // seen an explicit maxAge hint, set the maxAge to 0 (uncached) or the
+          // default if specified in the constructor. (Non-object fields by
+          // default are assumed to inherit their cacheability from their parents.
+          // But on the other hand, while root non-object fields can get explicit
+          // hints from their definition on the Query/Mutation object, if that
+          // doesn't exist then there's no parent field that would assign the
+          // default maxAge, so we do it here.)
+          if (
+            (targetType instanceof GraphQLObjectType ||
+              targetType instanceof GraphQLInterfaceType ||
+              !info.path.prev) &&
+            hint.maxAge === undefined
+          ) {
+            hint.maxAge = defaultMaxAge;
+          }
+
+          if (hint.maxAge !== undefined || hint.scope !== undefined) {
+            addHint(hints, info.path, hint);
+          }
+
+          info.cacheControl = {
+            setCacheHint: (hint: CacheHint) => {
+              addHint(hints, info.path, hint);
+            },
+            cacheHint: hint,
+          };
+        },
+      }),
+
+      responseForOperation() {
```

Comment:
@abernix Do you have any memory of why this is here? This runs before execution, so I don't see how the policy it calculates could ever be non-trivial. I don't see anything that obviously matches it on the "before" side of this PR. Tests pass without it (on my PR that refactors some stuff).